### ISSUE TITLE
SDXL model offload doesn't unload Unet if latent output is requested

### ISSUE THREAD

**Issue by Kubuxu** [huggingface/diffusers#4435]
If VEA decode is skipped (through `output_type = "latent"`), the `self.final_offload_hook.offload()` won't be called, and even if it was, it would trigger offload of the VAE, not the Unet. This can be worked around by calling `.to("cpu")` on the pipeline afterwards.

https://github.com/huggingface/diffusers/blob/816ca0048f7a38531a6f036299d2dd274abb5639/src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py#L844-L858

The simplest fix is to keep both Unet and VAE offload hooks and call the correct one depending on `output_type`.

**Comment by patrickvonplaten** [comment]
Good catch and very good idea on how to solve it - I think this should work! Would you like to open a PR here? :-) Also I wonder whether we can have a mechanism to just offload all models always by default when `cpu_model_offload` is enabled.

**Comment by sayakpaul** [comment]
Cc: @pcuenca as well.

**Comment by Qusijia** [comment]
I  want to get latentfeatures. However, I am a novice and would like to ask if there is any code that can be referenced? thanks！

**Comment by a-r-r-o-w** [comment]
I believe this is no longer a problem, so marking as closed 🤗
